provider/core/src/data_provider.rs

```diff
@@ -165,13 +183,13 @@ where
     ///
     /// Due to [compiler bug #84937](https://github.com/rust-lang/rust/issues/84937), call sites
     /// for this function may not compile; if this happens, use
-    /// [`try_from_rc_buffer_badly()`](Self::try_from_rc_buffer_badly) instead.
+    /// [`try_from_buffer_badly()`](Self::try_from_buffer_badly) instead.
     #[inline]
-    pub fn try_from_rc_buffer<E>(
-        rc_buffer: Rc<[u8]>,
+    pub fn try_from_buffer<B: Into<rc_type!()>, E>(
```

Comment:
Issue, here and elsewhere: Enabling/disabling features should not change the signature of any functions. If there's code that is built to rely on a feature being turned off, and another crate turns it on, you break things.

In other words, the change should be internal only, adding the `Send + Sync` to DataPayload but not changing any visible function signatures.

Not sure of the right solution right now, but thought I'd flag it first so we can discuss

Reply:
That's why I changed the signature to `B: Into<rc_type!()>`. `Rc` and `Arc` have the same `From` implementations.

Reply:
Issues with this:

1. We cannot assume that types that implement `Into<Rc<[u8]>>` necessarily implement `Into<Arc<[u8]>>` and vice-versa
2. Replying on `Into` is a footgun because it could cause the data to be cloned, and we really want to avoid this function having to clone any data

I suggested adding a `RcWrap` type, or basically an ICU4X-specific Rc that clients should use.